LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/LnsMessageType.cs

```diff
@@ -22,4 +25,40 @@ internal enum LnsMessageType
 
         */
     }
+
+    internal static class LnsMessageTypeExtensions
+    {
+        internal static bool TryParseLnsMessageType(string s, out LnsMessageType? lnsMessageType)
+        {
+            lnsMessageType = s switch
+            {
+                "version" => LnsMessageType.Version,
+                "router_config" => LnsMessageType.RouterConfig,
+                "jreq" => LnsMessageType.JoinRequest,
+                "updf" => LnsMessageType.UplinkDataFrame,
+                "dntxed" => LnsMessageType.TransmitConfirmation,
+                "dnmsg" => LnsMessageType.DownlinkMessage,
+                _ => null
+            };
+            return lnsMessageType is not null;
+        }
+
+        internal static LnsMessageType ParseAndValidate(string s, LnsMessageType? expectedType) =>
+            TryParseLnsMessageType(s, out var parsedType)
+               ? expectedType is null ^ parsedType == expectedType
+                   ? parsedType.Value
+                   : throw new ValidationException($"Input msgtype parsed as {parsedType}, but was expecting {expectedType}.")
+               : throw new FormatException($"Could not parse {s} as a valid {nameof(LnsMessageType)}.");
+
+        internal static string ToBasicStationString(this LnsMessageType lnsMessageType) => lnsMessageType switch
+        {
+            LnsMessageType.Version              => "version",
+            LnsMessageType.RouterConfig         => "router_config",
+            LnsMessageType.JoinRequest          => "jreq",
+            LnsMessageType.UplinkDataFrame      => "updf",
+            LnsMessageType.TransmitConfirmation => "dntxed",
+            LnsMessageType.DownlinkMessage      => "dnmsg",
+            _ => throw new SwitchExpressionException("Invalid internal state"),
```

Comment:
I think an argument error is more appropriate here:

```suggestion
            _ => throw new ArgumentOutOfRangeException(nameof(lnsMessageType), lnsMessageType, null),
```

The fact that we used a switch internally is an implementation detail.